src/sandstorm/web-session-bridge.c++

```diff
@@ -1248,6 +1254,59 @@ kj::Promise<void> WebSessionBridge::handleResponse(
       headers.set(tables.hContentSecurityPolicy, "default-src 'none'; sandbox");
 
       headers.set(tables.hAccessControlExposeHeaders, kj::strArray(exposedHeaders, ", "));
+    } else if(!allowLegacyRelaxedCSP) {
+      // Disallow loading of remote resources. Note the following:
+      //
+      // - Currently there are still exceptions for images and media, as these have
+      //   some legitimate use cases (e.g. embedding images in feeds in ttrss) and
```

Comment:
Not a priority for now, but are there any ideas for how to prevent information leaking when getting images? A malicious application could put something private in a GET string, but sometimes GET strings are legitimately required to return an image (requested size, etc).

Reply:
This has been talked about on IRC a bit, but what I want to try is:

- By default, block images & media.
- Use CSP's report-uri/report-to mechanisms to detect when grains try to load external images & media.
- When that happens, in the Sandstorm UI, pop up a thing that says "Images & media have been blocked for your privacy [load images & media]."
- If the user clicks the load images button, we refresh the contents of the iframe with a relaxed CSP.

This is an approach that I think will be familiar to people from experience with email clients.